app/controllers/application_controller.rb

```diff
@@ -35,7 +35,14 @@ def analytics
   end
 
   def create_user_event(event_type, user = current_user)
-    Event.create(user_id: user.id, event_type: event_type)
+    Event.transaction do
+      Event.create!(user_id: user.id, event_type: event_type)
+      # Keep total number of events per user under MAX_EVENTS_PER_USER
+      events = Event.where(user_id: user.id).order('created_at ASC')
+      if events.count > Event::MAX_EVENTS_PER_USER
```

Comment:
Its not really two different things in this case though since its the creation of the new event that is pushing the oldest one out.